server/cache_test.go

```diff
@@ -0,0 +1,415 @@
+package server
+
+import (
+	"testing"
+	"time"
+
+	"github.com/brave-intl/challenge-bypass-server/model"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type MockClock struct {
+	CurrentTime time.Time
+}
+
+func (m *MockClock) Now() time.Time {
+	return m.CurrentTime
+}
+
+func (m *MockClock) Advance(d time.Duration) {
+	m.CurrentTime = m.CurrentTime.Add(d)
+}
+
+func TestSimpleCache_SetDefault(t *testing.T) {
+	type testStruct struct {
+		A int
+		B string
+	}
+
+	tests := []struct {
+		name              string
+		defaultExpiration time.Duration
+		key               string
+		value             testStruct
+		wantStored        bool
+	}{
+		{
+			name:              "set_struct_value",
+			defaultExpiration: 5 * time.Second,
+			key:               "structkey",
+			value:             testStruct{A: 9, B: "b"},
+			wantStored:        true,
+		},
+		{
+			name:              "set_with_no_expiration",
+			defaultExpiration: 0,
```

Comment:
nit: this is redundant as `defaultExpiration` will take the default value of `0` so we dont need to set it